### ISSUE TITLE
readme

### ISSUE THREAD

**Issue by ololoe** [Drekin/win-unicode-console#35]
> This has nothing to do with encoding, since even sys.stdin.buffer.raw.readline() returns `b"?\n"` when entering `α` and there is no encoding under which sys.stdout.buffer.raw.write displays `α`.

Except you choose greek encoding? Or is it non-greek `α`? :)

**Comment by Drekin** [comment]
You are right, it is U+03B1 (GREEK SMALL LETTER ALPHA). It is an example of a character not in my charset, but which I can write on my keyboard. I naively used it as a general example of such character not considering Greek charset since I was viewing it as “mathematical alpha” since that is my usage.

The general point is that a codepage encodes only 256 characters, but you may enter 65536 characters to cmd.exe, which fails silently – e.g. greek β is converted to german ß, while α is turned into ?. And this compat conversion is done before it gets to sys.stdin decoder, so it is not caused directly by sys.stdin.encoding. So unless you start with Greek locale, just chaging sys.stdin.encoding won't help.

**Comment by ololoe** [comment]
I understand what you mean, but it's not clear if you read readme. So you should either change it to character that is not in any encoding (yes, math α and greek α are different symbols!) or to use symbols from different codepages in your example. Or maybe choose an asian glyph?

**Comment by ololoe** [comment]
Please add following into "There are additional issues on Python 2.":

The parameter of `python -c` or `py -2 -c` is non-unicode. `python -c "print 'α'"` will print `?`. There is no way to deal with it (or is there any?).

**Comment by Drekin** [comment]
Yes, the readme should be updated. Thank you for the suggestions.

Regarding the `-c` issue, it is caused by the processing of cmdline arguments by Python, so it affects only literals. For example `py -2 -c "print u'\u03b1'" works. I don't know if anything can be done about it. It depends on whether there is a way to fix the cmdline arguments during the interpret startup.

**Comment by ololoe** [comment]
Yes, it affects only literals, but this does not help if the language you are using uses non-latin character set.
